### PR TITLE
add hostlistener blog post

### PR DESCRIPTION
**Pull request by JWO719**

### CODE REVIEW

File: blog-posts/host-listener/host-listener.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+published: false
+title: "Ain't nobady needs HostListener"
```
Comment (tehfedaykin) [inline blog-posts/host-listener/host-listener.md]: "nobady" > "nobody"

File: blog-posts/host-listener/host-listener.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+cover_image: ""
+description: "In Angular event handling is often implemented using the hostListener decorator, eventhough it might not be the best fit for the problem. Have you consider more composable approaches?"
```
Comment (tehfedaykin) [inline blog-posts/host-listener/host-listener.md]: "eventhough" > "even though"

File: blog-posts/host-listener/host-listener.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+---
+published: false
+title: "Ain't nobody needs HostListener"
+cover_image: ""
+description: "In Angular event handling is often implemented using the hostListener decorator, even though it might not be the best fit for the problem. Have you consider more composable approaches?"
```
Comment (tehfedaykin) [inline blog-posts/host-listener/host-listener.md]: "Have you consider" > "Have you considered"

File: blog-posts/host-listener/host-listener.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+tags: angular, rxjs, javascript, typescript
+series:
+canonical_url:
+---
+
+Angular's `@hostListener` is well known within the community. Rather unknown are the problems this might have on runtime performance and general application architecture. In general, there are two main problems with the `hostListener` decorator.
```
Comment (tehfedaykin) [inline blog-posts/host-listener/host-listener.md]: "with the `hostListener`" > "with using the `hostListener`" 

This softens the language, taking the focus from the hostListener itself to the way a developer may attempt to use it.

File: blog-posts/host-listener/host-listener.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+1. It's not composable
+2. It can harm performance
+3. Lacks configuration options
```
Comment (tehfedaykin) [inline blog-posts/host-listener/host-listener.md]: for the list items - they should either all use "It/it's" or not use any contractions

File: blog-posts/host-listener/host-listener.md (hunk 4 above)
Comment (tehfedaykin) [inline blog-posts/host-listener/host-listener.md]: "two main problems" > there are three problems listed, or did you want to say "a few"?

Comment (JWO719) [reply blog-posts/host-listener/host-listener.md]: there used to be two at the beginning :D

**Pull request merged**